solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -244,20 +251,29 @@ contract WalletRegistry is
         _;
     }
 
-    constructor(
-        SortitionPool _sortitionPool,
-        IStaking _staking,
+    /// @dev Used to initialize immutable variables only, use `initialize` function
+    ///      for upgradable contract initialization on deployment.
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(SortitionPool _sortitionPool, IStaking _staking) {
+        sortitionPool = _sortitionPool;
+        staking = _staking;
+    }
+
+    /// @dev Initializes upgradable contract on deployment.
+    function initialize(
         EcdsaDkgValidator _ecdsaDkgValidator,
         IRandomBeacon _randomBeacon,
         ReimbursementPool _reimbursementPool
-    ) {
-        sortitionPool = _sortitionPool;
-        staking = _staking;
+    ) external initializer {
         randomBeacon = _randomBeacon;
         reimbursementPool = _reimbursementPool;
 
         // TODO: revisit all initial values
 
+        dkgResultSubmissionGas = 272_000;
+        dkgResultApprovalGasOffset = 63_000;
```

Comment:
I did some testing and I think we should actually increase it to `79k`. As a rule of thumb it is better for callers to be slightly net-positive rather than net-negative. If we lower it to `63k` then we have 3 tests failing and the callers would have to pay around 0.003ETH when the gas price is 200 gwei. However, if we increase this gas offset to 79k, then in a couple of scenarios we would need to increase the payout in `WalletRegistry.WalletCreation.test.ts`:
- line: 1784 from 0.002 to 0.0045
- line: 1636 from 0.001 to 0.0035

So it's a tradeoff, but we want to avoid for callers to be net-negative.

Reply:
b1b364fbca8f46fdca935d745c50608a606e1648

Reply:
In 8a49ccf7b41dca75aa36c2ba9bafbfb0c7601a87 I added verification of gas usage for transactions. It showed that value for `submitDkgResult` could be higher as the tests for `submitDkgResult` have higher gas consumption than `272 000`.